freeipa/src/main/java/com/sequenceiq/freeipa/service/freeipa/user/UserSyncService.java

```diff
@@ -243,20 +242,30 @@ private void internalSynchronizeUsers(String operationId, String accountId, Stri
 
             LogEvent logUserSyncEvent = fullSync ? LogEvent.FULL_USER_SYNC : LogEvent.PARTIAL_USER_SYNC;
             LOGGER.info("Starting {} for environments {} with operationId {} ...", logUserSyncEvent, environmentCrns, operationId);
-            LogEvent logRetrieveUmsEvent = fullSync ? LogEvent.RETRIEVE_FULL_UMS_STATE : LogEvent.RETRIEVE_PARTIAL_UMS_STATE;
-            LOGGER.debug("Starting {} for environments {} ...", logRetrieveUmsEvent, environmentCrns);
-            Map<String, UmsUsersState> envToUmsStateMap = umsUsersStateProvider
-                    .getEnvToUmsUsersStateMap(accountId, actorCrn, environmentCrns, userCrnFilter, machineUserCrnFilter, requestId);
-            LOGGER.debug("Finished {}.", logRetrieveUmsEvent);
+
+            Map<String, Future<SyncStatusDetail>> statusFutures;
+
+            if (userSyncFilter.getDeletedWorkloadUser().isEmpty()) {
+                LogEvent logRetrieveUmsEvent = fullSync ? LogEvent.RETRIEVE_FULL_UMS_STATE : LogEvent.RETRIEVE_PARTIAL_UMS_STATE;
+                LOGGER.debug("Starting {} for environments {} ...", logRetrieveUmsEvent, environmentCrns);
+                Map<String, UmsUsersState> envToUmsStateMap = umsUsersStateProvider
+                        .getEnvToUmsUsersStateMap(accountId, actorCrn, environmentCrns, userSyncFilter.getUserCrnFilter(),
+                                userSyncFilter.getMachineUserCrnFilter(), requestId);
+                LOGGER.debug("Finished {}.", logRetrieveUmsEvent);
+                statusFutures = stacks.stream()
+                        .collect(Collectors.toMap(Stack::getEnvironmentCrn,
+                                stack -> asyncSynchronizeStack(stack, envToUmsStateMap.get(stack.getEnvironmentCrn()), umsEventGenerationIds, fullSync,
+                                        operationId, accountId)));
+            } else {
+                String deletedWorkloadUser = userSyncFilter.getDeletedWorkloadUser().get();
+                LOGGER.debug("Starting {} for environments {} and deleted user {} ...", LogEvent.USER_SYNC_DELETE, environmentCrns, deletedWorkloadUser);
```

Comment:
I don't see a matching Finished log for USER_SYNC_DELETE. I think you need to move this logging into internalSynchronizeStackForDeleteUser